uvtools/plot.py

```diff
@@ -471,6 +472,428 @@ def omni_view_gif(filenames, name='omni_movie.gif'):
+        and ``times`` is not provided.
+    time_or_lst: str, optional
+        Either "time" or "lst". Used to specify whether the time axis should be
+        in JD or LST. If ``data`` is an array-like object and only one of ``times``
+        or ``lsts`` is provided, then this parameter is ignored.
+    mode: str, optional
+        Plotting mode to use; must be a mode accepted by :func:`data_mode`. Default
+        is "log", which plots the base-10 logarithm of the absolute value of the data.
+    ax: :class:`plt.Axes` instance, optional
+        :class:`plt.Axes` object to use for plotting the waterfall. If not provided,
+        then a new :class:`plt.Figure` object and :class:`plt.Axes` instance is created.
+    figsize: tuple of int, optional
+        Length-2 tuple specifying figure dimensions in inches. Ignored if ``ax`` 
+        is provided.
+    dpi: int, optional
+        Dots per inch to be used in creating the figure. Ignored if ``ax`` is provided.
+    aspect: str or float, optional
+        Aspect ratio of the plot. Should be either "equal", "auto", or a number.
+        Default is to use "auto".
+    fontsize: float, optional
+        Font size for labels, in points.
+    draw_colorbar: bool, optional
+        Whether to draw a colorbar. Default is to draw a colorbar.
+    cmap: str or :class:`plt.cm.colors.Colormap` instance, optional
+        Colormap to use for plotting the waterfall. Default is to choose a colormap
+        appropriate for the plotting mode chosen ("twilight" for plotting phases,
+        and "inferno" otherwise).
+    dynamic_range: float, optional
+        Number of orders of magnitude of dynamic range to plot. For example, setting
+        ``dynamic_range=5`` limits the colorbar to range from the maximum value to
+        five orders of magnitude below the maximum. If ``mode=="phs"``, then this
+        parameter is ignored.
+    fft_axis: int or str, optional
+        Axis over which to perform a Fourier transform. May be specified with one
+        of three strings ("time", "freq", "both") or one of three integers (0, 1,
+        -1), where the integers map to the axes specified by the strings. Default 
+        is to not perform a Fourier transform over any axis.
+    freq_taper: str, optional
+        Taper to use when performing a Fourier transform along the frequency axis.
+        Must be one of the tapers supported by :func:`dspec.gen_window`.
+    freq_taper_kwargs: dict, optional
+        Keyword arguments to use in generating the taper for the frequency axis.
+    time_taper: str, optional
+        Taper to use when performing a Fourier transform along the time axis.
+        Must be one of the tapers supported by :func:`dspec.gen_window`.
+    time_taper_kwargs: dict, optional
+        Keyword arguments to use in generating the taper for the time axis.
+
+    Returns
+    -------
+    fig or ax: :class:`plt.Figure` instance or :class:`plt.Axes` instance
```

Comment:
Its a bit odd and potentially less than ideal to have the return type change. Since all figures have both a fig and an array of axes, why not just have the output mimic the output of  fig, axes = plt.subplots()

Reply:
Noted. It seemed straightforward to me that passing an `Axes` object (by setting `ax=...` in the function call) would get an `Axes` object returned, and a figure otherwise, though I do see how it can be confusing. I can go ahead and make the change.